impl/src/planning/subplanners.rs

```diff
@@ -439,225 +426,208 @@ impl<'planner> CrateSubplanner<'planner> {
     license::get_license_from_str(licenses_str)
   }
 
-  fn _produce_deps(&self, names: &DependencyNames) -> Result<DependencySet> {
-    let build_dep_names = &names.build_dep_names;
-    let dev_dep_names = &names.dev_dep_names;
-    let normal_dep_names = &names.normal_dep_names;
-    let aliased_dep_names = &names.aliased_dep_names;
-
-    let mut dep_set = DependencySet {
-      build_deps: Vec::new(),
-      build_proc_macro_deps: Vec::new(),
-      proc_macro_deps: Vec::new(),
-      dev_deps: Vec::new(),
-      normal_deps: Vec::new(),
-      aliased_deps: Vec::new(),
-    };
+  /// Generates the set of dependencies for the contained crate.
+  fn produce_deps(&self) -> Result<DepProduction> {
+    let mut dep_production = DepProduction::new();
 
     let all_skipped_deps = self
       .crate_settings
       .iter()
       .flat_map(|pkg| pkg.skipped_deps.iter())
       .collect::<HashSet<_>>();
 
-    for dep_id in &self.node.dependencies {
+    // This wonderful part of the metadata gives us both renames and targets.
+    //
+    // Irritatingly its still a little tricky to detect renames fully, we need this to finally
+    // deduce the aliases - see `is_renamed`.
+    //
+    // If rust-lang/cargo#7289 gets solved then a lot of the left-over rename detection code can
```

Comment:
```suggestion
    // If https://github.com/rust-lang/cargo/issues/7289 gets solved then a lot of the left-over rename detection code can
```

This is optional but I personally prefer full links unless it's to the same repo as the project.